microhapulator/profile.py

```diff
@@ -62,30 +62,30 @@ def initialize(self):
             "markers": dict(),
         }
 
-    def haplotypes(self):
+    def haploindexes(self):
         hapids = set()
         for markerid, markerdata in self.data["markers"].items():
             for alleledata in markerdata["genotype"]:
-                if "haplotype" in alleledata and alleledata["haplotype"] is not None:
-                    hapids.add(alleledata["haplotype"])
+                if "index" in alleledata and alleledata["index"] is not None:
+                    hapids.add(alleledata["index"])
         assert sorted(hapids) == sorted(range(len(hapids)))
         return hapids
 
     def markers(self):
-        return set(list(self.data["markers"]))
+        return set(self.data["markers"])
 
-    def alleles(self, markerid, haplotype=None):
+    def haplotypes(self, markerid, index=None):
```

Comment:
The changes to this core module weren't strictly necessary, but certainly relevant to the motivations for this PR and a good time to address some accumulated technical debt.